site/en/docs/extensions/mv3/screen_capture/index.md

```diff
@@ -33,28 +31,97 @@ If called within a content script, recording will automatically end when the use
 
-In the future, we may support passing a media stream ID to an
-[offscreen document][offscreen-documents] so recording can happen in the background and more easily
-persist across navigations. We are collecting feedback in the
-[chromium-extensions mailing list][feedback-mailing-list].
+In your service worker:
 
-{% endAside %}
+```js
+chrome.action.onClicked.addListener(async (tab) => {
+  const existingContexts = await chrome.runtime.getContexts({});
+
+  const offscreenDocument = existingContexts.find(
+    (c) => c.contextType === 'OFFSCREEN_DOCUMENT'
+  );
+
+  // If an offscreen document is not already open, create one.
+  if (!offscreenDocument) {
+    // Create an offscreen document.
+    await chrome.offscreen.createDocument({
+      url: 'offscreen.html',
+      reasons: ['USER_MEDIA'],
+      justification: 'Recording from chrome.tabCapture API',
+    });
+  }
+
+  // Get a MediaStream for the active tab.
+  const streamId = await chrome.tabCapture.getMediaStreamId({
+    targetTabId: tab.id
+  });
 
-To record audio and video across navigations, you can open an extension page in a new tab or window,
-and directly obtain a stream. Set the `targetTabId` property to capture the correct tab.
+  // Send the stream ID to the offscreen document to start recording.
+  chrome.runtime.sendMessage({
+    type: 'start-recording',
+    target: 'offscreen',
+    data: streamId
+  });
+});
+```
 
-In your popup:
+Then, in your offscreen document:
+
+```js
+chrome.runtime.onMessage.addListener(async (message) => {
+  if (message.target !== 'offscreen') return;
+  
+  if (message.type === 'start-recording') {
+    const media = await navigator.mediaDevices.getUserMedia({
+      audio: {
+        mandatory: {
+          chromeMediaSource: "tab",
+          chromeMediaSourceId: message.data,
+        },
+      },
+      video: {
+        mandatory: {
+          chromeMediaSource: "tab",
+          chromeMediaSourceId: message.data,
+        },
+      },
+    });
+
+    // Continue to play the captured audio to the user.
+    const output = new AudioContext();
+    const source = output.createMediaStreamSource(media);
+    source.connect(output.destination);
+
+    // TODO: Do something to recording the MediaStream.
+  }
+});
+```
+
+For a full example, see the [Tab Capture - Recorder][recorder-sample] sample.
+
+### Recording audio and video in a new tab {: #audio-and-video-new-tab }
+
+Prior to Chrome 116, it was not possible to use the [`chrome.tabCapture`][tabcapture] API in a
+service worker or to consume a stream ID created by that API in an offscreen document. Both of these
+are requirements for the approach above.
+
+Instead, you can open an extension page in a new tab or window, and directly obtain a stream. Set
+the `targetTabId` property to capture the correct tab.
+
+Start by opening an extension page (perhaps in your popup or service worker):
 
 ```js
 chrome.windows.create({ url: chrome.runtime.getURL("recorder.html") });
```

Comment:
I'd rather keep this actually so it works everywhere (maybe a developer has a button on the page to start recording and their content script calls this). It feels nice to have a snippet that works everywhere especially since the documentation isn't prescribing a particular place.

Reply:
Oh I see. My suggestion was based on the intro paragraph:

> "Start by opening an extension page (perhaps in your popup or service worker):"